eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Binary/TxWitsSpec.hs

```diff
@@ -0,0 +1,238 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Alonzo.Binary.TxWitsSpec (spec) where
+
+import Cardano.Ledger.Alonzo.Core
+import Cardano.Ledger.Alonzo.Scripts
+import Cardano.Ledger.Alonzo.TxWits (AlonzoTxWits (..))
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Binary (DeserialiseFailure (..))
+import Cardano.Ledger.Binary.Decoding
+import Cardano.Ledger.Binary.Encoding
+import Cardano.Ledger.Keys
+import Cardano.Ledger.Plutus.Language
+import Data.List (isPrefixOf)
+import qualified Data.Map as Map
+import qualified Data.Maybe as Maybe (mapMaybe)
+import Data.Typeable
+import Test.Cardano.Ledger.Alonzo.Arbitrary
+import Test.Cardano.Ledger.Common
+
+spec ::
+  forall era.
+  ( AlonzoEraScript era
+  , Script era ~ AlonzoScript era
+  ) =>
+  Proxy era ->
+  Spec
+spec pxy = do
```

Comment:
Let's use consistently `AllowAmbiguousTypes` for `spec`.
```suggestion
spec ::
  forall era.
  ( AlonzoEraScript era
  , Script era ~ AlonzoScript era
  ) =>
  Spec
spec = do
```

Reply:
Nice. Also did it for `CostModelsSpec`, removed tons of noise, thanks.